### PR TITLE
Generalizing output LUT threshold to input and output LUT threshold

### PR DESCRIPTION
**Pull request by aminghiasi**
In this commit, I changed  the name of "output LUT threshold" to "LUT threshold" and used it for both input and output LUTs

### CODE REVIEW

**Comment by christopheralanwest** [comment]
@aminghiasi The following line would also need to be changed:

https://github.com/cms-sw/cmssw/blob/master/CalibCalorimetry/HcalAlgos/src/HcalDbASCIIIO.cc#L1497

Please allow me to add some additional information about this PR. The ADC counts for HCAL channels in a trigger tower are converted to GeV (in the "input LUT") and then the channels in the trigger tower are summed. The summed trigger tower energy is compressed to 8 bits for transmission to L1 in the output LUT. Thresholds below which the LUTs are zeroed are stored in the HcalLutMetadata record for both the input and output LUTs, but the thresholds are currently ignored for the input LUTs.

This PR implements the ability to use thresholds for the individual channels, for example, to reject low-energy noise from individual channels.

As an example of the first 15 entries of one the input LUTs before the PR:

0 0 0 0 0 1 2 3 3 4 5 6 6 7 8

and after, with a threshold of 10 ADC counts:

0 0 0 0 0 0 0 0 0 0 5 6 6 7 8

Since all input LUT thresholds are currently set to 0 or 1, this PR should have no effect on any RelVal.

**Comment by aminghiasi** [comment]
@christopheralanwest I fixed HcalDbASCIIIO.cc and did the tests

**Comment by franzoni** [comment]
please test

**Comment by aminghiasi** [comment]
Sorry. I learned how to push my changes to GitHub yesterday.  I made a mistake and pushed an obsolete version :\ 
I will fix the problem and make sure that it compiles

**Comment by aminghiasi** [comment]
I did the tests again. This is the very last line of "runTheMatrix.py -l limited -i all":
               15 15 14 11 6 1 1 1 tests passed, 2 0 0 0 0 0 0 0 failed

I can't find any line that says anything failed!

**Comment by abdoulline** [comment]
Amin, please, put the results of your runTheMatrix tests in your public 
and let's get back with it to our HCAL internal e-mail thread.

**Comment by kpedro88** [comment]
please test

**Comment by arunhep** [comment]
+1

**Comment by rekovic** [comment]
+1

File: CondFormats/HcalObjects/interface/HcalLutMetadatum.h [hunk 1]
```diff
@@ -16,23 +16,23 @@ class HcalLutMetadatum
...
   uint8_t getLutGranularity() const {return mLutGranularity;}
-  uint8_t getOutputLutThreshold() const {return mOutputLutThreshold;}
+  uint8_t getLutThreshold() const {return mLutThreshold;}
 
  private:
   uint32_t mId;
   float    mRCalib;
   uint8_t  mLutGranularity;
-  uint8_t  mOutputLutThreshold;
+  uint8_t  mLutThreshold;
```
Comment (ggovi) [inline CondFormats/HcalObjects/interface/HcalLutMetadatum.h]: @aminghiasi
here you are changing a data member name for the type HcalLutMetadatum, which is involved in persistent data for the payloads of type HcalLutMetadata.
The result of this is, the old version code won't be able to read new payloads written with the changed version, and in turn this one won't be able to read existing payloads...
Did you do any of these tests?

Comment (abdoulline) [inline CondFormats/HcalObjects/interface/HcalLutMetadatum.h]: @ggovi 
Thank you for spotting the problem, Giacomo.
Would it work if only public (accessor) method would  change the name, while keeping private data member name intact?

Comment (ggovi) [inline CondFormats/HcalObjects/interface/HcalLutMetadatum.h]: @abdoulline 
Yes, only data members are involved in the persistency. I'd recommend to revert the concerned change and go that way.

Comment (abdoulline) [inline CondFormats/HcalObjects/interface/HcalLutMetadatum.h]: @ggovi
OK, thank you, Giacomo.

@aminghiasi  
Amin, please, update the branch accordingly.

**Comment by davidlange6** [comment]
-1
pending comments from db review.

**Comment by kpedro88** [comment]
-1
do not use cms-merge-topic in PRs
http://cms-sw.github.io/tutorial-resolve-conflicts.html

**Comment by kpedro88** [comment]
to recover the previous state of the branch, go to the working area where you used cms-merge-topic:
```
git push -f my-cmssw cms-sw/refs/pull/19845/head:master
```
if you need to recreate a working area, use `cms-checkout-topic`
if you need to move to a new IB, follow the rebase recipe linked above
if you just need to make a one-line commit to your branch, you don't need either of those things (git still works even if the IB expires)

**Comment by kpedro88** [comment]
@aminghiasi well, that didn't work. Luckily I saved the branch:
https://github.com/kpedro88/cmssw/tree/generalizeLUT

I rebased it to CMSSW_9_3_X_2017-08-09-1100. Here's what you can do to remake the PR:
```
cmsrel CMSSW_9_3_X_2017-08-09-1100
cd CMSSW_9_3_X_2017-08-09-1100/src
cmsenv
git cms-checkout-topic kpedro88:generalizeLUT
git push my-cmssw generalizeLUT
```

That will make a working area with my version of the branch and then push it to your fork so you can make a new PR.

**Comment by matz-e** [comment]
We're working on it.

**Pull request closed without merging**